### PR TITLE
feat: add support for setting initial solutions with SCIP

### LINKED ISSUE #55: Give a solution to the solver
Hi again!

Most, if not all, of the solver allow to give a feasible solution to the solver. I have a case where I have a solution that is optimal, but I don’t know the optimality. The solver find as a bound the objective value, but continue to search as it doesn’t find the solution. For some solver, it can also improve the resolution.

Can we add this feature? Any remark? How to manage the case of the solver that doesn’t support it (I suspect the executable interface doesn’t always allow this)?a

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Title says it all.

Depends on #71 being merged first.

When these changes are applied, two out of the three large open-source solvers CBC, HiGHs, and SCIP support initial solutions. Adding this feature for HiGHS needs support in the underlying bindings. This is tracked in https://github.com/rust-or/highs/issues/21. Proprietary solvers would also need updated bindings first. Nevertheless, I'd argue that this

closes #55.

### CODE REVIEW

File: src/solvers/scip.rs [hunk 1]
```diff
@@ -154,6 +154,17 @@ impl SolverModel for SCIPProblem {
     }
 }
 
+impl WithInitialSolution for SCIPProblem {
+    fn with_initial_solution(self, solution: impl IntoIterator<Item = (Variable, f64)>) -> Self {
+        let sol = self.model.create_sol();
+        for (var, val) in solution {
+            sol.set_val(Rc::clone(&self.id_for_var[&var]), val);
+        }
+        self.model.add_sol(sol);
```
Comment (KnorpelSenf) [inline src/solvers/scip.rs:163]: I wonder if we should check this result somehow, or surface it to the callsite

Comment (KnorpelSenf) [reply src/solvers/scip.rs:163]: I'll at least do
```suggestion
        self.model.add_sol(sol).expect("could not set solution");
```

**Comment by KnorpelSenf** [comment]
@lovasoa could you publish a new release after this is reviewed and merged?

**Comment by KnorpelSenf** [comment]
@lovasoa hang on for a few more minutes

**Comment by KnorpelSenf** [comment]
@lovasoa how would we do this for HiGHS with the new changes in https://github.com/rust-or/highs/pull/22? Do we now have to derive the dual solution from the given initial solution, or is this something we want to delegate to library consumers?

**Comment by lovasoa** [comment]
I honestly have no idea what's best here

**Comment by KnorpelSenf** [comment]
It turns out that we can just pass `None` so I guess we don't really have to worry about it.

**Pull request merged**